examples/pipelines/windows-installer/README.md

```diff
@@ -0,0 +1,60 @@
+# Windows Installer Pipeline
+
+Downloads a Windows Source ISO into a PVC and installs Windows into a new base PVC.
+It uses Windows Sysprep to automatically install the system. 
+The pipeline can be customized to support different installation requirements. 
+
+The artifact of this pipeline is a PVC (backed by DataVolume) with a custom Windows installation. 
+
+## Prerequisites
+
+- KubeVirt `v0.39.0`
+- Tekton Pipelines `v0.19.0`
+
+### Obtain Windows ISO download URL
+
+1. Go to https://www.microsoft.com/en-us/software-download/windows10ISO. You can also obtain a server edition for evaluation at https://www.microsoft.com/en-us/evalcenter/evaluate-windows-server-2019.
+2. Fill in the edition and language and go to the download page.
+3. Right-click on the 64-bit download button and copy the download link. The link should be valid for 24 hours.
+4. Initialize a WIN_URL variable that will be used to create a DataVolume which will download this ISO into a PVC.
+
+```bash
+# Real URL can look differently
+WIN_URL="https://software-download.microsoft.com/db/Win10_20H2_v2_English_x64.iso..."
+```
+
+### Prepare autounattend.xml ConfigMap
+
+1. Supply, generate or use the default autounattend.xml.
+2. Replace the default example autounattend.xml with your own in `windows-installer-pipeline.yaml`. 
+   You can also store the config inside a secret (VM definition requires changes in that case).
+   Different autounattend.xml can be also passed in the Pipeline parameters when creating a PipelineRun.
+
+## Pipeline Description
+
+```
+  create-source-dv --- create-vm-from-manifest --- wait-for-vm --- cleanup-vm (finally)
+                    |
+    create-base-dv --
+```
+
+1. `create-source-dv` task downloads a Windows source ISO into a PVC called `windows-10-source-*`.
+2. `create-base-dv` task creates empty PVC for new windows installation called `windows-10-base-*`.
+3. `create-vm-from-manifest` task creates a VM called `windows-installer-*` 
+   from the empty PVC and with `windows-10-base-*` PVC attached as a CD-ROM.
+4. `wait-for-vm` [WIP] task waits until the VM shutdowns.
+5. `cleanup-vm` [WIP] deletes the installer VM and ISO PVC. 
+6.  The output artifact will be the `windows-10-base-*` PVC with the Windows installation.
+
+## How to run
+
+```bash
+WIN_URL="https://software-download.microsoft.com/db/Win10_20H2_v2_English_x64.iso..."
+kubectl apply -f windows-installer-pipeline.yaml
+sed 's!DOWNLOAD_URL!'"$WIN_URL"'!g' windows-installer-pipelinerun.yaml | kubectl create -f -
+```
+
+## Possible Optimizations
+
+- Windows source ISO can be downloaded and cached/stored in a PVC. 
+  So the subsequent PipelineRuns don't have to download the same ISO multiple times.
```

Comment:
Can you provide the optimization right now? The pipeline should better work without Internet connectivity.

Reply:
I would prefer for the pipeline to be as simple as possible so one can run the example quickly.  But I can describe the extra steps here in this chapter.

Reply:
fixed